<a href="https://colab.research.google.com/github/shahzarhusain/python_libraries/blob/main/NPS_problem.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

We want to check the NPS for an Airbnb property. NPS i.e. Net promoter score is a metric based on the ratings provided by the customers for the likelihood that they will recommned the product within their circle.

**NPS = % of promoters - % of detractors**

The criteria ratings for promoter and detractor varies from organisation to organisation. Usually, for the scale of 1-10:
*   1-6 - Detractors
*   7-8 - Neutral
*   9-10 - Promoters

The NPS can vary from -100 % to 100%.

In [3]:
import numpy as np

In [1]:
# importing the data file
!gdown 1c0ClC8SrPwJq5rrkyMKyPn80nyHcFikK

Downloading...
From: https://drive.google.com/uc?id=1c0ClC8SrPwJq5rrkyMKyPn80nyHcFikK
To: /content/survey.txt
100% 2.55k/2.55k [00:00<00:00, 8.33MB/s]


In [4]:
score = np.loadtxt('survey.txt', dtype = 'int')
score

array([ 7, 10,  5, ...,  5,  9, 10])

The dataset is an one dimensional array consist of Ratings only.



In [5]:
score.ndim

1

In [6]:
score.shape

(1167,)

In [7]:
len(score)

1167

In [8]:
# let's verify the range of ratings
print(score.min())
print(score.max())

1
10


In [9]:
detractors = score[score <= 6]

In [10]:
promoters = score[score >=9 ]

In [14]:
total_ratings = score.shape[0]
total_ratings

1167

In [17]:
num_detractors = detractors.shape[0]
(num_detractors/total_ratings)*100

28.449014567266495

In [18]:
num_promoters = promoters.shape[0]
(num_promoters/total_ratings)*100

52.185089974293064

In [12]:
nps = (num_promoters - num_detractors)/total_ratings *100

In [13]:
nps

23.736075407026565

We got an NPS of 23.75, which seems a good score but there is still room for improvement as we can see around 28% of detractors.